### PR TITLE
Add bcl_convert component, including unit tests

### PR DESCRIPTION
**Pull request by tverbeiren**
## Description

bcl_convert component for demultiplexing a bcl directory.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)
- [ ] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)
- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`
- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

File: src/bcl_convert/test.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# print final message
+echo ">>> Test finished successfully"
+
+echo ">> Execute with additional arguments and verify output"
+
+$meta_executable \
+  --bcl_input_directory "$test_data_dir/iseq-DI" \
+  --sample_sheet "$test_data_dir/sample_sheet.csv" \
+  --output_directory fastq1 \
+  --bcl_only_matched_reads true \
```
Comment (dorien-er) [inline src/bcl_convert/test.sh:53]:
```suggestion
  --bcl_only_matched_reads \
```

File: src/bcl_convert/test.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  --bcl_num_compression_threads 1 \
+  --no_lane_splitting false \
```
Comment (dorien-er) [inline src/bcl_convert/test.sh:55]:
```suggestion
```

**Review by dorien-er — CHANGES_REQUESTED** [review]

**Comment by dorien-er** [comment]
lgtm :)

File: src/bcl_convert/script.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -eo pipefail
+
+$(which bcl-convert) \
+  ${par_bcl_input_directory:+ --bcl-input-directory "$par_bcl_input_directory"} \
```
[the remark this answers is not in the document]
Comment (tverbeiren) [reply src/bcl_convert/script.sh:6]: Good point! Mandatory arguments should not be checked anymore.

**Comment by tverbeiren** [comment]
Thanks for the comments! I tried to align the boolean arguments to how bcl-convert is handling them, hence the `boolean` rather than `boolean_true` or `boolean_false`.

**Pull request merged**